src/components/SeatingPlan.vue

```diff
@@ -0,0 +1,325 @@
+      tooltip.style.top = `${(e as MouseEvent).clientY + 10}px`;
+      tooltip.classList.remove('hidden');
+    }
+  } else if (!slot) {
+    hoveredTeamSlot.value = null;
+    hoveredTeamSlotName.value = null;
+
+    // hide the tooltip
+    const tooltip = document.getElementById('tooltip');
+    if (tooltip) {
+      tooltip.classList.add('hidden');
+    }
+  }
+};
+
+const handleClick = (index: number) => {
+  const [x, y] = getCoordinates(index);
+
+  if (!isTournamentSeat(index)) return;
+
+  // find the seatslot containing the hovered seat
+  const slot = props.tournament.seatslots.find(
+    (seatslot) => seatslot.seats.some((seat) => seat.x === x && seat.y === y),
+  );
+  if (slot) {
+    // find the team occupying the hovered seat
+    const team = props.tournament.teams.find((t) => (t as unknown as TeamDeref).seat_slot === slot.id);
+    if (!team) {
+      selectedTeamSlot.value = slot.id;
+      showModal.value = true;
+    }
+  }
+};
+
+const validateModal = async () => {
+  showModal.value = false;
+  await patch_team(
+    (props.team as Team)?.id,
+    {
+      seat_slot: selectedTeamSlot.value,
+    },
+  );
+};
+
+const closeModal = () => {
+  showModal.value = false;
+  selectedTeamSlot.value = null;
+};
+
+</script>
+
+<template>
+  <section id="seating">
+    <div v-if="tournament?.event.seats" class="flex flex-col items-center justify-center">
+      <div
+        class="m-2 w-screen overflow-x-auto"
+        :style="{
+          maxWidth: `${(maxX + 1) * 32}px`,
+        }"
+      >
+        <div
+          class="p-4"
+          :style="{
+            width: `${(maxX + 1) * 32}px`,
+          }"
+        >
+          <h2 class="mb-4 text-2xl font-bold">
+            Placement des équipes pour : {{ tournament.event.name }}
+          </h2>
+          <div
+            id="tooltip"
+            class="z-1 fixed hidden max-w-60 items-center justify-center overflow-hidden rounded-lg border border-black bg-gray-600 p-2 text-white"
+          >
+            <div v-if="hoveredTeamSlotName" class="truncate">
+              Équipe : <strong class="text-blue-500">
+                {{ hoveredTeamSlotName }}
+              </strong>
+            </div>
+            <template v-else>
+              Places libres
+            </template>
+          </div>
+          <div
+            class="grid"
+            TournamentDeref
+            :style="{
+              gridTemplateColumns: `repeat(${maxX + 1}, minmax(0, 1fr))`,
+              gridTemplateRows: `repeat(${maxY + 1}, minmax(0, 1fr))`,
+            }"
+            @mouseleave="handleHover(-1, $event)"
+            @focusout="handleHover(-1, $event)"
+          >
+            <div
+              v-for="(_, index) in (maxY + 1) * (maxX + 1)"
+              :key="index"
+              :class="[
+                'flex size-8 items-center justify-center text-xs',
+              ]"
+              @click="team ? handleClick(index) : handleHover(index, $event)"
```

Comment:
Team is not always null, it depend from which view it was imported. The slot selection is only possible from the team view.
When imported from the tournament view, it is indeed null because this component is used to display the global seating map.
When imported from the team view, it has a team props which is the team selected.

Reply:
My bad, I thought we can choose our placement from both pages